iceoryx_utils/test/moduletests/test_cxx_function_ref.cpp

```diff
@@ -60,6 +61,17 @@ ComplexType returnComplexType(ComplexType foo)
     return foo;
 }
 
+int SameSignature(function_ref<void(void)> callback)
+{
+    callback();
+    return -1;
+}
+
+int SameSignature(function_ref<int(int)> callback)
```

Comment:
I know it is a bit pedantic but could you please also add unit tests for

- ```int SameSignature(function_ref<int(void)> callback)```
- ```int SameSignature(function_ref<void(int)> callback)```
- ```int SameSignature(function_ref<int(int, int)> callback)```

Reply:
Added. However, `int SameSignature(function_ref<void(int)> callback)` would be ambiguous to `int SameSignature(function_ref<int(int)> callback)` as the return value is not part of the signature, so this one is left out.

Reply:
You are right.